request_handler/request_handler.go

```diff
@@ -90,23 +90,24 @@ func auditTrail(e *spade.Event) string {
 	return fmt.Sprintf("[%d] %s", e.ReceivedAt.Unix(), e.Uuid)
 }
 
-func getIpFromHeader(headerKey string, header http.Header) net.IP {
-	clientIp := header.Get(headerKey)
-	comma := strings.Index(clientIp, ",")
-	if comma > -1 {
-		clientIp = clientIp[:comma]
+func parseLastForwarder(header string) net.IP {
+	var clientIp string
+	comma := strings.LastIndex(header, ",")
+	if comma > -1 && comma < len(header)+1 {
+		clientIp = header[comma+1:]
+	} else {
+		clientIp = header
```

Comment:
comma < len(header)+1 really should just never be hit; it would require the ELB to see an x-forwarded-for and either append a malformed ip or *not* modify it? It is here just so we don't explode; I don't think we have a defined behavior in this case.